psdk_wrapper/src/modules/camera.cpp

```diff
@@ -1190,11 +1308,24 @@ PSDKWrapper::camera_download_file_by_index_cb(
   {
     RCLCPP_INFO(get_logger(), "Download file by index successful.");
     response->success = true;
+  }
+
+  return_code = DjiCameraManager_ReleaseDownloaderRights(index);
+  if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+  {
+    RCLCPP_ERROR(get_logger(),
+                 "Release Downloader Rights failed, error code: %ld.",
+                 return_code);
+    response->success = false;
+    return;
+  }
+  else
+  {
+    RCLCPP_INFO(get_logger(), "Release Downloader Rights successful.");
     return;
   }
 }
 
-// TODO(@lidiadltv): Not working. Debug potential issue
 void
 PSDKWrapper::camera_delete_file_by_index_cb(
```

Comment:
This function the same, I see some synergies with the previous function which would make the code smaller and more readable. For instance, the ObtainRights and ReleaseRights could be a generic method that both functions benefit from it and all the boilerplate about return codes is centralized into  a single function. Look for those similarities